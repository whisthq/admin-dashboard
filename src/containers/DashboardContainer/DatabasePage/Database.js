import React, { Component } from 'react'
import { Tab, Tabs, TabList, TabPanel } from 'react-tabs'
import { connect } from 'react-redux'

import 'react-tabs/style/react-tabs.css'
import 'static/App.css'

import {
    fetchUserActivity,
    fetchUserTable,
    deleteUser,
    changePage,
} from 'actions/index'

import VMTable from 'containers/DashboardContainer/DatabasePage/components/VMTable'
import UserTable from 'containers/DashboardContainer/DatabasePage/components/UserTable'
import CustomerTable from 'containers/DashboardContainer/DatabasePage/components/CustomerTable'
import DiskTable from 'containers/DashboardContainer/DatabasePage/components/DiskTable'

class Dashboard extends Component {
    constructor(props) {
        super(props)
        this.state = {
            loaded: false,
            userTableFetched: false,
        }
    }

    componentDidMount() {
        this.props.dispatch(fetchUserActivity(false))
        this.props.dispatch(changePage('dashboard'))
    }

    componentDidUpdate(prevProps) {
        if (
            this.props.access_token &&
            this.props.userTable.length === 0 &&
            !this.state.userTableFetched
        ) {
            this.props.dispatch(fetchUserTable())
            this.setState({ userTableFetched: true })
        }
    }

    deleteUser = (user) => {
        this.props.dispatch(deleteUser(user))
    }

    render() {
        return (
            <div
                style={{
                    position: 'relative',
                    bottom: 10,
                }}
            >
                <div
                    style={{
                        fontSize: 45,
                        fontWeight: 'bold',
                        marginBottom: 60,
                        width: 275,
                        color: '#111111',
                    }}
                >
                    DASHBOARD
                </div>
                <div
                    style={{
                        display: 'flex',
                        marginTop: 5,
                    }}
                >
                    <div
                        style={{
                            display: 'block',
                            width: '100%',
                            position: 'relative',
                            bottom: 36,
                        }}
                    >
                        <Tabs>
                            <TabList
                                style={{
                                    textAlign: 'left',
                                    border: 'none',
                                    fontSize: 16,
                                }}
                            >
                                <Tab>Cloud PCs</Tab>
                                <Tab>Disks</Tab>
                                <Tab>Users</Tab>
                                <Tab>Customers</Tab>
                            </TabList>
                            <TabPanel style={{ paddingTop: 20 }}>
                                <div>
                                    <VMTable />
                                </div>
                            </TabPanel>
                            <TabPanel style={{ paddingTop: 20 }}>
                                <div>
                                    <DiskTable />
                                </div>
                            </TabPanel>
                            <TabPanel style={{ paddingTop: 20 }}>
                                <div>
                                    <UserTable />
                                </div>
                            </TabPanel>
                            <TabPanel style={{ paddingTop: 20 }}>
                                <CustomerTable openModal={() => null} />
                            </TabPanel>
                        </Tabs>
                    </div>
                </div>
            </div>
        )
    }
}

function mapStateToProps(state) {
    return {
        vm_info: state.AccountReducer.vm_info
            ? state.AccountReducer.vm_info
            : [],
        vmsUpdated: state.AccountReducer.vmsUpdated,
        activityFetched: state.AccountReducer.activityFetched,
        userActivity: state.AccountReducer.userActivity
            ? state.AccountReducer.userActivity
            : [],
        userTable: state.AccountReducer.userTable
            ? state.AccountReducer.userTable
            : [],
        usersUpdated: state.AccountReducer.usersUpdated,
        access_token: state.AccountReducer.access_token,
        login_attempts: state.AccountReducer.login_attempts,
    }
}

export default connect(mapStateToProps)(Dashboard)
