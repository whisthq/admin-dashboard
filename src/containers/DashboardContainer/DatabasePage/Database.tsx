import React from 'react'
import { Tab, Tabs, TabList, TabPanel } from 'react-tabs'
import { connect } from 'react-redux'

import 'react-tabs/style/react-tabs.css'
import '../../../static/App.css'

import { changePage } from '../../../actions/index'

import VMTable from './components/VMTable'
import UserTable from './components/UserTable'
import CustomerTable from './components/CustomerTable'
import DiskTable from './components/DiskTable'

export class Dashboard extends React.Component<any, any> {
    componentDidMount() {
        this.props.dispatch(changePage('dashboard'))
    }

    render() {
        return (
            <div
                style={{
                    position: 'relative',
                    bottom: 10,
                }}
            >
                <div
                    style={{
                        fontSize: 45,
                        fontWeight: 'bold',
                        marginBottom: 60,
                        width: 275,
                        color: '#111111',
                    }}
                >
                    DASHBOARD
                </div>
                <div
                    style={{
                        display: 'flex',
                        marginTop: 5,
                    }}
                >
                    <div
                        style={{
                            display: 'block',
                            width: '100%',
                            position: 'relative',
                            bottom: 36,
                        }}
                    >
                        <Tabs>
                            <TabList
                                style={{
                                    textAlign: 'left',
                                    border: 'none',
                                    fontSize: 16,
                                }}
                            >
                                <Tab>Cloud PCs</Tab>
                                <Tab>Disks</Tab>
                                <Tab>Users</Tab>
                                <Tab>Customers</Tab>
                            </TabList>
                            <TabPanel style={{ paddingTop: 20 }}>
                                <div>
                                    <VMTable />
                                </div>
                            </TabPanel>
                            <TabPanel style={{ paddingTop: 20 }}>
                                <div>
                                    <DiskTable />
                                </div>
                            </TabPanel>
                            <TabPanel style={{ paddingTop: 20 }}>
                                <div>
                                    <UserTable />
                                </div>
                            </TabPanel>
                            <TabPanel style={{ paddingTop: 20 }}>
                                <CustomerTable openModal={() => null} />
                            </TabPanel>
                        </Tabs>
                    </div>
                </div>
            </div>
        )
    }
}

function mapStateToProps(state: any) {
    return {
        access_token: state.AccountReducer.access_token,
    }
}

export default connect(mapStateToProps)(Dashboard)
