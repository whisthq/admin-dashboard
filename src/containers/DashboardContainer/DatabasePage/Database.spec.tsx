import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { Database } from './Database'

Enzyme.configure({ adapter: new Adapter() })

/*
The Database is basically a table with tabs that have various differnet properties. There should be a <Tabs> with
four TabPanels and four <Tab> elements in a TabList. There is one for the VMTable, one for the DiskTable, one for
the UserTable, and one for the CustomerTable. We must make sure that each of these show up. Everything else is 
just styling.

On component mount we simply need user activity to be fetched and the page to be changed (not sure about user activity
being fetched). When the component updates we need fetches for the user table or something like that and the state to
be set properly. These are not clear if they are necessary since it appears that the tables handle it so we won't
provide test bodies.
*/

describe('<Database />', () => {
    test('ensure componentDidMount fetches necessary data and dispatchs page changes', () => {
        // TODO
    })
    test('ensure componentDidUpdate fetches necessary data and dispatches state information', () => {
        // TODO
    })
    // TODO it may be nice to test the helper method to delete users, but I'm not sure its used

    it('displays a tablist with four tabs and then four tab panels all in a <Tabs>', () => {
        // really
    })

    it('contains each of the components: VMTable, DiskTable, UserTable, CustomerTable', () => {
        // really
    })
})

/*
All the Database's components are tested very shallowly. Tests that are ommitted are those that may check that
the right props are passed to the antd table, or that the right width/colors are generated based on different
parameters. These are much more complicated, and we may want to hold off until we have reason to do so.
*/