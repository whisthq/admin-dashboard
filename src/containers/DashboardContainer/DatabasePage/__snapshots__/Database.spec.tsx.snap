// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Database /> renders correctly 1`] = `
<div
  style={
    Object {
      "bottom": 10,
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "color": "#111111",
        "fontSize": 45,
        "fontWeight": "bold",
        "marginBottom": 60,
        "width": 275,
      }
    }
  >
    DASHBOARD
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "marginTop": 5,
      }
    }
  >
    <div
      style={
        Object {
          "bottom": 36,
          "display": "block",
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <Tabs
        defaultFocus={false}
        defaultIndex={null}
        forceRenderTabPanel={false}
        selectedIndex={null}
      >
        <TabList
          className="react-tabs__tab-list"
          style={
            Object {
              "border": "none",
              "fontSize": 16,
              "textAlign": "left",
            }
          }
        >
          <Tab
            className="react-tabs__tab"
            disabledClassName="react-tabs__tab--disabled"
            focus={false}
            id={null}
            panelId={null}
            selected={false}
            selectedClassName="react-tabs__tab--selected"
          >
            Cloud PCs
          </Tab>
          <Tab
            className="react-tabs__tab"
            disabledClassName="react-tabs__tab--disabled"
            focus={false}
            id={null}
            panelId={null}
            selected={false}
            selectedClassName="react-tabs__tab--selected"
          >
            Disks
          </Tab>
          <Tab
            className="react-tabs__tab"
            disabledClassName="react-tabs__tab--disabled"
            focus={false}
            id={null}
            panelId={null}
            selected={false}
            selectedClassName="react-tabs__tab--selected"
          >
            Users
          </Tab>
          <Tab
            className="react-tabs__tab"
            disabledClassName="react-tabs__tab--disabled"
            focus={false}
            id={null}
            panelId={null}
            selected={false}
            selectedClassName="react-tabs__tab--selected"
          >
            Customers
          </Tab>
        </TabList>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          selectedClassName="react-tabs__tab-panel--selected"
          style={
            Object {
              "paddingTop": 20,
            }
          }
        >
          <div>
            <Connect(VMTable) />
          </div>
        </TabPanel>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          selectedClassName="react-tabs__tab-panel--selected"
          style={
            Object {
              "paddingTop": 20,
            }
          }
        >
          <div>
            <Connect(DiskTable) />
          </div>
        </TabPanel>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          selectedClassName="react-tabs__tab-panel--selected"
          style={
            Object {
              "paddingTop": 20,
            }
          }
        >
          <div>
            <Connect(UserTable) />
          </div>
        </TabPanel>
        <TabPanel
          className="react-tabs__tab-panel"
          forceRender={false}
          selectedClassName="react-tabs__tab-panel--selected"
          style={
            Object {
              "paddingTop": 20,
            }
          }
        >
          <Connect(CustomerTable)
            openModal={[Function]}
          />
        </TabPanel>
      </Tabs>
    </div>
  </div>
</div>
`;
